client/gutenberg/extensions/slideshow/component.js

```diff
@@ -0,0 +1,154 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { createRef, Children, Component } from '@wordpress/element';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+
+import Swiper from 'swiper';
+import 'swiper/dist/css/swiper.min.css';
+
+export class Slideshow extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			images: [],
+			imageHeight: 400,
+		};
+		this.slideshowRef = createRef();
+		this.btnNextRef = createRef();
+		this.btnPrevRef = createRef();
+		this.paginationRef = createRef();
+	}
+	render() {
+		const { children, className } = this.props;
+		const { imageHeight } = this.state;
+		const classNames = classnames( className, 'swiper-container' );
+		return (
+			<div className={ classNames } ref={ this.slideshowRef }>
+				<div className="swiper-wrapper">
+					{ Children.map( children, child => {
+						if ( ! child.props.children ) {
+							return null;
+						}
+						const image = this.getChildByDataAttribute( child, 'data-is-image' );
+						if ( ! image ) {
+							return null;
+						}
+						const { src, alt } = image.props;
+						const figcaption = this.getChildByDataAttribute( child, 'data-is-caption' );
+						const caption = figcaption ? figcaption.props.children : null;
+						const style = {
+							backgroundImage: `url(${ src })`,
+							height: imageHeight,
+						};
+						return (
+							<div className="swiper-slide">
+								<div className="slide-background atavist-cover-background-color" />
+								<div className="wp-block-slideshow-image-container" style={ style } title={ alt } />
+								{ caption && <p className="slideshow-slide-caption">{ caption }</p> }
+							</div>
+						);
+					} ) }
+				</div>
+				<div className="swiper-pagination swiper-pagination-white" ref={ this.paginationRef } />
+				<div className="swiper-button-prev swiper-button-white" ref={ this.btnPrevRef } />
+				<div className="swiper-button-next swiper-button-white" ref={ this.btnNextRef } />
+			</div>
+		);
+	}
+	componentDidMount() {
+		this.buildImageMetadata( this.buildSwiper );
+	}
+	componentDidUpdate( prevProps ) {
+		const { swiperInstance } = this.state;
+		const { align, effect, children } = this.props;
+		if ( children !== prevProps.children ) {
```

Comment:
Could this check get us in trouble? This is for referential equality -- [`React.children`](https://reactjs.org/docs/react-api.html#reactchildren) is an opaque data structure, but probably some kind of object, meaning that `this.props.children` and `prevProps.children` could be identical in terms of contents, whereas their references might not be (i.e. the `!==` check would yield `true`) -- which would mean we'd call `this.buildImageMetadata()` more often than necessary...